### PR TITLE
Kaito UX Overhaul

### PR DESCRIPTION
**Pull request by tejhan**
This PR introduces an interactive menu for selecting & generating kaito workspaces, alongside subtle UI tweaks throughout.

- New page for generating KAITO workspaces

Screenshot 2024-09-17 171952

- Refactored UI to allow accessibility across varying color themes
- Created post-installation button that opens new kaito model page

image

- Added kaito models webview test suite

### CODE REVIEW

**Comment by tejhan** [comment]
Vsix file for testing -> [vscode-aks-tools-1.4.9-kaito-ux-update.zip](https://github.com/user-attachments/files/17047961/vscode-aks-tools-1.4.9-kaito-ux-update.zip)

**Review by ReinierCC — DISMISSED** [review]
LGTM, tested VSIX on end as well.

File: webview-ui/src/Kaito/KaitoModels.module.css [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    font-size: 1.1rem;
+    font-style: normal;
+    font-weight: 700;
+    line-height: normal;
+    cursor: pointer;
+    border: none;
+    background: none;
+    display: flex;
+    align-items: center;
+    font-family: var(--vscode-font-family);
```
Comment (hsubramanianaks) [inline webview-ui/src/Kaito/KaitoModels.module.css:46]: I think do we change font families ? we can keep all fonts same?

Comment (tejhan) [reply webview-ui/src/Kaito/KaitoModels.module.css:46]: Oh yeah, when developing it, I realized that for whatever reason, the <button> element doesn't inherit the default vscode font, it uses Arial by default, so I just added this here to make sure it uses the same font as everything else.

File: src/commands/aksKaito/aksKaitoCreateCRD.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        return;
+    }
+
+    const clusterName = clusterNode.result.name;
+    const armId = clusterNode.result.armId;
+    const subscriptionId = clusterNode.result.subscriptionId;
+    const resourceGroupName = clusterNode.result.resourceGroupName;
+
+    const panel = new KaitoModelsPanel(extension.result.extensionUri);
+    const dataProvider = new KaitoModelsPanelDataProvider(clusterName, subscriptionId, resourceGroupName, armId);
```
Comment (Tatsinnit) [inline src/commands/aksKaito/aksKaitoCreateCRD.ts:44]: 💡 this is nice and simple, just one minor observation I have is that when I was playing with the U/X the toggle panel could be if we select one the other gets hidden, but that just minor thing which I thought I will mention.

**Review by Tatsinnit — APPROVED** [review]
<3 Thanks for this, this U/X looks solid to me. Thanks for this.

There is movement of code here and I wonder if we can test this post merge. https://github.com/tejhan/vscode-aks-tools/pull/1

File: webview-ui/src/Kaito/Kaito.module.css [hunk 3]
```diff
@@ -1,78 +1,91 @@
...
-    color: var(--Base-Base-01, #fff);
-    font-size: 1.625rem;
-    font-style: normal;
-    font-weight: 600;
-    line-height: normal;
-}
-
 .subHeader {
-    width: 41rem;
+    width: 100vw;
```
Comment (hsubramanianaks) [inline webview-ui/src/Kaito/Kaito.module.css:2]: can we do width's as well in rem? just to make sure it works in all vscode themes? please

File: webview-ui/src/Kaito/Kaito.module.css [hunk 4]
```diff
@@ -1,78 +1,91 @@
...
-    margin-bottom: 0.625rem;
+    margin-top: 1.4rem;
+    margin-bottom: 0.4rem;
+    color: var(--vscode-editor-foreground, #fff);
 }
 
 .architectureSubHeader {
-    width: 40.6rem;
-    color: var(--Dark-Base-03, #ccc);
+    width: 100vw;
```
Comment (hsubramanianaks) [inline webview-ui/src/Kaito/Kaito.module.css:26]: same comment as above.

File: webview-ui/src/Kaito/Kaito.module.css [hunk 5]
```diff
@@ -1,78 +1,91 @@
...
+    color: var(--vscode-editor-foreground, #fff);
     font-size: 0.8125rem;
     font-style: normal;
     font-weight: 700;
     line-height: normal;
     margin-left: 0.625rem;
 }
+
+.spacer {
+    margin-top: 500px;
```
Comment (hsubramanianaks) [inline webview-ui/src/Kaito/Kaito.module.css:76]: px can be converted into rem's please.

File: webview-ui/src/Kaito/KaitoModels.module.css [hunk 6]
```diff
@@ -0,0 +1,165 @@
+body {
+    color: var(--vscode-editor-foreground, #fff);
+}
+
+.mainDiv {
+    margin-bottom: 2rem;
+}
+
+.modelsHeader {
+    font-size: 26px;
```
Comment (hsubramanianaks) [inline webview-ui/src/Kaito/KaitoModels.module.css:10]: same here px as rem's.

File: webview-ui/src/Kaito/KaitoModels.module.css [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    font-style: normal;
+    font-weight: 600;
+    line-height: normal;
+    margin: 1rem 0 0.5rem 0;
+}
+
+.subHeader {
+    width: 100vw;
```
Comment (hsubramanianaks) [inline webview-ui/src/Kaito/KaitoModels.module.css:18]: same here rem instead of vw.

File: webview-ui/src/Kaito/KaitoModels.tsx [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+                                viewBox="0 0 16 16"
+                                fill="none"
+                            >
+                                <path
+                                    fillRule="evenodd"
+                                    clipRule="evenodd"
+                                    d="M7.97612 10.0719L12.3334 5.71461L12.9521 6.33333L8.28548 11L7.66676 11L3.0001 6.33333L3.61882 5.71461L7.97612 10.0719Z"
+                                    className={styles.arrowPath}
+                                />
+                            </svg>
```
Comment (hsubramanianaks) [inline webview-ui/src/Kaito/KaitoModels.tsx:102]: can we extract this svc similar to DevTestIcon to it's own file?

**Review by hsubramanianaks — APPROVED** [review]
looks good to me except few minor changes.

File: webview-ui/src/Kaito/KaitoModels.tsx [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+inference:
+  preset:
+    name: "${name}"`;
+        vscode2.postGenerateCRDRequest({ model: yaml });
+        return;
+    }
+
+    return (
+        <div className={styles.mainDiv}>
+            <h2>Create a KAITO Workspace</h2>
```
Comment (Tatsinnit) [inline webview-ui/src/Kaito/KaitoModels.tsx:70]: 👍 Important note: Lets add clusterName here, if user has many, its hard to distinguish between screen, also add same cluster name for the createWorkspace as well. Thanks heaps!

Comment (hsubramanianaks) [reply webview-ui/src/Kaito/KaitoModels.tsx:70]: I agree I was looking for which cluster did I enabled Kaito lol.. :)

**Comment by tejhan** [comment]
This PR is superseded by PR #963.

**Pull request closed without merging**